example/lib/interactive_features_example.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:flutter/material.dart';
+import 'package:mapbox_maps_flutter/mapbox_maps_flutter.dart';
+import 'example.dart';
+
+class InteractiveFeaturesExample extends StatefulWidget implements Example {
+  @override
+  final Widget leading = const Icon(Icons.map);
+  @override
+  final String title = 'Interactive Features';
+  @override
+  final String? subtitle = 'Click to select buildings, long-click to unselect';
+
+  @override
+  State<StatefulWidget> createState() => InteractiveFeaturesState();
+}
+
+class InteractiveFeaturesState extends State<InteractiveFeaturesExample> {
+  InteractiveFeaturesState();
+  MapboxMap? mapboxMap;
+
+  _onMapCreated(MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+    mapboxMap.style;
+  }
+
+  _onTap(context) async {
+    // Define the geometry to query, in this case the point where the user clicked.
+    var clicked = await mapboxMap?.pixelForCoordinate(context.point);
+    var renderedQueryGeometry =
+        RenderedQueryGeometry.fromScreenCoordinate(clicked!);
+
+    // Define the featureset to query. In this case "buildings", which is defined in the
+    // Standard Experimental style.
+    var featureset =
+        FeaturesetDescriptor(featuresetId: "buildings", importId: "basemap");
+
+    // Query the featureset for the geometry
+    var queriedFeatures = await mapboxMap?.queryRenderedFeaturesForFeatureset(
+        geometry: renderedQueryGeometry, featureset: featureset);
+    var featuresetFeature = queriedFeatures?.first;
+
+    if (featuresetFeature != null) {
+      // Define the state to set for the feature, in this case highlighting
+      // Set that featurestate on that featuresetFeature
+      Map<String, Object?> state = {
+        "highlight": true,
+      };
+      mapboxMap?.setFeatureStateForFeaturesetFeature(featuresetFeature, state);
```

Comment:
Unfortunately, no. I agree that they are quite long, but I'm not sure how to shorten them without losing meaning. WDYT, @evil159 & @maios? 

QRF:

- queryRenderedFeatures(_RenderedQueryGeometry geometry, RenderedQueryOptions options)
  - This is the existing method  
- queryRenderedFeaturesForTargets(_RenderedQueryGeometry geometry, List<FeaturesetQueryTarget> targets)
- queryRenderedFeaturesForFeatureset( _RenderedQueryGeometry geometry, FeaturesetDescriptor featureset, String? filter)
- queryRenderedFeaturesInViewport(FeaturesetDescriptor featureset, String? filter)

QSF:
- querySourceFeatures(
      String sourceId, SourceQueryOptions options)
  - Existing method
- querySourceFeaturesForFeatureset(FeaturesetQueryTarget target)
  - Perhaps this could be querySourceFeaturesForTargets

Set/Get FeatureState: 
- set(get)FeatureState(
      String sourceId, String? sourceLayerId, String featureId, String state)
  - Existing method
- set(get)FeatureStateForFeaturesetDescriptor(
    FeaturesetDescriptor featureset, FeaturesetFeatureId featureId)
- set(get)FeatureStateForFeaturesetFeature(
    FeaturesetFeature feature, Map<String, Object?> state)

We could maybe drop "Featureset" from these ones 🤔